src/ui/lib/components/DeployedPackageMenu/connect-deployed-package-dialog.svelte

```diff
@@ -0,0 +1,127 @@
+<!-- 
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+ -->
+<script lang="ts">
+	import { IconButton, List, Typography, type SSX } from '@ui';
+	import type { DeployedPackage } from '$lib/api-types';
+	import ButtonDense from '../button-dense.svelte';
+	import ZarfDialog from '../zarf-dialog.svelte';
+	import { onDestroy, onMount } from 'svelte/internal';
+	import { Packages } from '$lib/api';
+	import { updateConnections } from '$lib/store';
+
+	// Props
+	export let pkg: DeployedPackage;
+	export let toggleDialog: () => void;
+
+	// Locals
+	let open: boolean;
+	let currentWindow: Window;
+	let selectedConnection = '';
+	let errMessage: string = '';
+	const connections: string[] = Object.keys(pkg.data.metadata?.connectStrings || {}) || [];
+
+	const listSSX: SSX = {
+		$self: {
+			maxHeight: '100px',
+			overflowY: 'scroll',
+			padding: '0px',
+			'& > li': {
+				display: 'flex',
+				alignItems: 'center',
+				padding: '0px',
+			},
+			'& .icon-button': {
+				width: '38px !important',
+				height: '38px !important',
+				'&:hover': {
+					background: 'var(--shades-primary-16p)',
+				},
+				color: 'var(--primary)',
+				'& .material-symbols-outlined': {
+					fontSize: '17px',
+				},
+			},
+		},
+	};
+
+	async function connectPackage(): Promise<void> {
+		try {
+			const connection = await Packages.connect(pkg.name, selectedConnection);
+			currentWindow.open(connection.url, '_blank');
+			await updateConnections();
+			toggleDialog();
+		} catch (err: any) {
+			errMessage = err.message;
+		}
+	}
+
+	onMount(() => {
+		currentWindow = window;
+	});
+
+	onDestroy(() => {
+		errMessage = '';
+	});
+
+	$: failedToConnect = errMessage !== '';
+	$: titleText =
+		(failedToConnect && `Failed to connect to ${selectedConnection}`) || 'Connect to Resource';
+	$: {
+		if (open && connections.length > 0) {
+			selectedConnection = connections[0];
+		}
+	}
+</script>
+
+<ZarfDialog bind:open bind:toggleDialog {titleText} happyZarf={!failedToConnect}>
+	<svelte:fragment>
+		{#if !failedToConnect}
+			<Typography variant="body1" color="text-secondary-on-dark">
+				Select which resource you would like Zarf to connect to. Zarf will create a tunnel and open
+				the connection in a new tab
+			</Typography>
+			<List ssx={listSSX}>
+				{#each connections as connection}
```

Comment:
There is no state for no connections available currently.

Reply:
![Screenshot 2023-05-03 at 5 09 54 PM](https://user-images.githubusercontent.com/86802655/236077211-d07761a7-2201-43ae-99e3-4adaf43ae38f.png)